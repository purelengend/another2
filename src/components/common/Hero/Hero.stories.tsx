import type { Meta } from '@storybook/react'
import { MemoryRouter } from 'react-router-dom'
import Hero from './index'

const meta: Meta = {
  title: 'Practice Two/Hero Component',
  component: Hero,
  decorators: [
    Story => (
      <MemoryRouter>
        <div
          style={{
            backgroundColor: '#532EE7',
          }}
        >
          <Story />
        </div>
      </MemoryRouter>
    ),
  ],
  parameters: {
    layout: 'top',
  },
  tags: ['autodocs'],
}

const heroSectionContentSample = {
  imageUrl: 'https://assets.nicepagecdn.com/d2cc3eaa/3159880/images/yttyy.jpg',
  link: '#',
  title: 'We help businesses and brands grow',
  description:
    'Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident',
}

const Template = () => (
  <Hero
    imageUrl={heroSectionContentSample.imageUrl}
    buttonHref={heroSectionContentSample.link}
    title={heroSectionContentSample.title}
    description={heroSectionContentSample.description}
  />
)

export const Primary = Template.bind({})

export default meta
