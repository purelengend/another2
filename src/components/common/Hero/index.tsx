import React, { memo } from 'react'
import { Box, Flex, Image, Text, Button, Heading } from '@chakra-ui/react'
import { HeroImageProps, HeroDetailProps, HeroProps } from '@types'
import { Link } from 'react-router-dom'

const HeroImage: React.FC<HeroImageProps> = ({ imageUrl }) => {
  return (
    <Box
      className="hero-image-wrapper"
      paddingTop={{ base: '60px', xl: '0' }}
      height={{ base: '436px', md: '915px', xl: '600px' }}
    >
      <Image
        src={imageUrl}
        alt="This is the image of hero section"
        className="hero-image"
        width={{ xl: '410px' }}
        height={{ base: '452px', md: '931px', xl: '606px' }}
        objectFit="cover"
        boxShadow="5px -6px 20px 0 rgba(0,0,0,0.25)"
      />
    </Box>
  )
}

const HeroDetail: React.FC<HeroDetailProps> = ({
  title,
  description,
  buttonHref,
}) => {
  return (
    <Box
      className="hero-detail"
      color="textPrimary"
      width={{ xl: '470px' }}
      padding={{ xl: '0 50px' }}
    >
      <Heading
        as="h2"
        className="hero-title"
        fontFamily="Oswald-Regular"
        fontWeight="600"
        fontSize={{ base: '34px', md: '60px' }}
      >
        {title}
      </Heading>
      <Text
        className="hero-description"
        margin="30px 0"
        fontSize="17px"
        fontStyle="italic"
        lineHeight="1.8"
      >
        {description}
      </Text>
      <Button
        as={Link}
        to={buttonHref}
        className="hero-button"
        letterSpacing={1}
        padding={{ base: '18px 57px 19px', md: '21px 55px 21px' }}
        borderRadius="10px"
        backgroundColor="secondary"
        color="textSecondary"
        fontFamily="OpenSans-Semibold"
        height={{ base: 'full', xl: '62px' }}
        transition="all 0.3s ease-in-out"
        _hover={{
          backgroundColor: 'primary',
        }}
      >
        Learn more
      </Button>
    </Box>
  )
}

const Hero: React.FC<HeroProps> = ({
  imageUrl,
  title,
  description,
  buttonHref,
}) => {
  return (
    <Flex
      as="section"
      className="hero-section"
      flexDirection="column"
      position="relative"
      paddingTop="68px"
      minHeight={{ base: '1120px', md: '1568px', xl: '834px' }}
      marginBottom="20px"
    >
      <Flex
        justifyContent="flex-end"
        marginBottom="185px"
        marginRight={{ base: '-10px', md: '-50px' }}
      >
        {/* Background element */}
        <Box
          backgroundColor="secondary"
          height={{ base: '197px', md: '220px' }}
          width={{
            base: 'calc(((100% - 340px) / 2) + 318px)',
            md: 'calc(((100% - 720px) / 2) + 520px)',
            xl: 'calc(((100% - 940px) / 2) + 520px)',
          }}
        />
      </Flex>
      <Flex
        justifyContent="flex-start"
        marginLeft={{ base: '-10px', md: '-50px' }}
      >
        {/* Background element */}
        <Box
          backgroundColor="secondary"
          height="197px"
          width={{
            base: 'calc(((100% - 340px) / 2) + 318px)',
            md: 'calc(((100% - 720px) / 2) + 520px)',
            xl: 'calc(((100% - 940px) / 2) + 520px)',
          }}
        />
      </Flex>
      <Flex
        className="hero-content"
        width={{ base: '340px', md: '720px', xl: '940px' }}
        height={{ xl: '600px' }}
        position="absolute"
        left={{
          base: 'calc(((100% - 340px) / 2))',
          md: 'calc(((100% - 720px) / 2))',
          xl: 'calc(((100% - 940px) / 2))',
        }}
        flexDirection={{ base: 'column-reverse', xl: 'row' }}
        padding="30px"
        borderWidth="1px"
        backgroundColor="textSecondary"
        marginTop="48px"
      >
        <HeroImage imageUrl={imageUrl} />
        <HeroDetail
          title={title}
          description={description}
          buttonHref={buttonHref}
        />
      </Flex>
    </Flex>
  )
}

export default memo(Hero)
