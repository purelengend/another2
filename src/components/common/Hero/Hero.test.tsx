import { render, screen } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import Hero from '.'

describe('Hero Component', () => {
  const testProps = {
    imageUrl: 'test-image-url',
    title: 'Test Title',
    description: 'Test Description',
    buttonHref: '/test-link',
  }

  it('matches snapshot', () => {
    const { container } = render(
      <MemoryRouter>
        <Hero {...testProps} />
      </MemoryRouter>
    )
    expect(container).toMatchSnapshot()
  })

  it('renders HeroImage and HeroDetail components', () => {
    render(
      <MemoryRouter>
        <Hero {...testProps} />
      </MemoryRouter>
    )

    const heroImageElement = screen.getByAltText(
      'This is the image of hero section'
    )
    expect(heroImageElement).toBeInTheDocument()

    const heroTitleElement = screen.getByText('Test Title')
    const heroDescriptionElement = screen.getByText('Test Description')
    const learnMoreButtonElement = screen.getByText('Learn more')
    expect(heroTitleElement).toBeInTheDocument()
    expect(heroDescriptionElement).toBeInTheDocument()
    expect(learnMoreButtonElement).toBeInTheDocument()
  })
})
