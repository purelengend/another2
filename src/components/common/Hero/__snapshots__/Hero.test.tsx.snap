// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero Component matches snapshot 1`] = `
<div>
  <section
    class="hero-section css-1i968st"
  >
    <div
      class="css-1t0sl2q"
    >
      <div
        class="css-wc2o4l"
      />
    </div>
    <div
      class="css-1b25ukq"
    >
      <div
        class="css-h5qkms"
      />
    </div>
    <div
      class="hero-content css-1n8ugxz"
    >
      <div
        class="hero-image-wrapper css-s8jnda"
      >
        <img
          alt="This is the image of hero section"
          class="hero-image css-1xuawgx"
          src="test-image-url"
        />
      </div>
      <div
        class="hero-detail css-1w2bpi"
      >
        <h2
          class="chakra-heading hero-title css-wgs5kc"
        >
          Test Title
        </h2>
        <p
          class="chakra-text hero-description css-slxfkz"
        >
          Test Description
        </p>
        <a
          class="chakra-button hero-button css-18yvqgz"
          href="/test-link"
        >
          Learn more
        </a>
      </div>
    </div>
  </section>
</div>
`;
